.github/workflows/plugins_react_tests.yml

```diff
@@ -60,17 +60,35 @@ jobs:
       - name: Install Foreman npm dependencies
         run: npm ci --no-audit
         working-directory: ${{ github.workspace }}/projects/foreman
-      - name: Checkout ${{ matrix.plugin }}
+      - name: Checkout ${{ matrix.plugin.repo }}
         uses: actions/checkout@v4
         with:
-          repository: ${{ matrix.plugin }}
+          repository: ${{ matrix.plugin.repo }}
           path: ${{ github.workspace }}/projects/plugin
-      - name: Generate ${{ matrix.plugin }} npm dependencies package-lock
+      - name: Set up plugin in Foreman
+        run: |
+          echo "gemspec name: '${{ matrix.plugin.name }}', path: '${{ github.workspace }}/projects/plugin'" > "bundler.d/${{ matrix.plugin.name }}.local.rb"
+          if [ -d ${{ matrix.plugin.name }}/gemfile.d ] ; then
+            cat ${{ matrix.plugin.name }}/gemfile.d/*.rb >> bundler.d/${{ matrix.plugin.name }}.local.rb
+          fi
+        working-directory: ${{ github.workspace }}/projects/foreman
+      - name: "Set up Ruby ${{ matrix.ruby }}"
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+          working-directory: ${{ github.workspace }}/projects/foreman
+      - name: Generate ${{ matrix.plugin.name }} npm dependencies package-lock
         run: npm install --package-lock-only --no-audit --legacy-peer-deps
         working-directory: ${{ github.workspace }}/projects/plugin
-      - name: Install ${{ matrix.plugin }} npm dependencies
+      - name: Install ${{ matrix.plugin.name }} npm dependencies
         run: npm ci --no-audit --legacy-peer-deps
         working-directory: ${{ github.workspace }}/projects/plugin
-      - name: Run ${{ matrix.plugin }} tests
-        run: npm test
-        working-directory: ${{ github.workspace }}/projects/plugin
+      - run: sudo apt-get update
+      - run: sudo apt-get -qq -y install build-essential libcurl4-openssl-dev zlib1g-dev libpq-dev libvirt-dev
```

Comment:
given `setup-ruby` with `bundler-cache: true` succeeds above, these *seem* not to be necessary.

so you can either drop them, or move before the `setup-ruby` and see what you can drop from the `BUNDLE_WITHOUT` list (realistically tho, I don't expect plugins to need libvirt integration… so… drop?)